adagucserverEC/CMakeEProfile.cpp

```diff
@@ -655,13 +665,12 @@ int EProfileUniqueRequests::plotHeightRetrieval(CDrawImage *drawImage, CDFObject
         }
       }
     }
-  } else {
-    CDBError("plotHeightRetrievalVariable == NULL");
   }
   return 0;
 }
 
-int EProfileUniqueRequests::drawEprofile(CDrawImage *drawImage, CDF::Variable *variable, size_t *, size_t *count, EProfileUniqueRequests::Request *, CDataSource *dataSource) {
+int EProfileUniqueRequests::drawEprofile(CDrawImage *drawImage, CDF::Variable *variable, size_t *, size_t *count, EProfileUniqueRequests::Request *, CDataSource *dataSource,
+                                         CT::string *eProfileJson) {
```

Comment:
It feels strange how this function (and the ones higher up) takes both drawImage and eProfileJson, but ends up filling only one of them. But I am not sure this can be easily avoided, without major refactoring.

Reply:
True, well spotted. The idea is to avoid code duplication. Split them in their own functions with their own responsibility is probably more clear but that would take a major refactor. So I suggest to leave as is until we get a concrete use case for this.